brouter-map-creator/src/main/java/btools/mapcreator/OsmCutter.java

```diff
@@ -165,6 +192,93 @@ private void generatePseudoTags(Map<String, String> map) {
   }
 
 
+// Ess Bee : NEW function to add the new tags (estimated_noise_class , river, forest...)
+
+  private void generateSpecialTags(long osm_id, Map<String, String> map) {
+
+    if (jdbcurl == null) return;
+
+    // is the database allready connected?
+    if (conn == null) {
+
+      String sql_all_tags = "SELECT *  from all_tags where losmid = ?";
+
+      System.err.println("OsmCutter start connection to the database........" + jdbcurl);
+
+      try {
+        conn = DriverManager.getConnection(jdbcurl);
+        psAllTags = conn.prepareStatement(sql_all_tags);
+
+        System.err.println("OsmCutter connect to the database ok........");
+
+      } catch (SQLException g) {
+        System.err.format("Osm Cutter SQL State: %s\n%s\n", g.getSQLState(), g.getMessage());
+        System.exit(1);
+        return;
+      } catch (Exception f) {
+        f.printStackTrace();
+        System.exit(1);
+        return;
+      }
+    }
+
+    for (Map.Entry<String, String> e : map.entrySet()) {
+      if (e.getKey().equals("highway")) {
+        cntHighways = cntHighways + 1;
+
+        try {
+
+          psAllTags.setLong(1, osm_id);
+
+          // process the results
+          rsBrouter = psAllTags.executeQuery();
+
+          if (rsBrouter.next()) {
+
+            cntWayModified = cntWayModified + 1;
+            if (rsBrouter.getString("noise_class") != null) {
+              map.put("estimated_noise_class", rsBrouter.getString("noise_class"));
+              cntNewNoise = cntNewNoise + 1;
+            }
+            if (rsBrouter.getString("river_class") != null) {
+              map.put("estimated_river_class", rsBrouter.getString("river_class"));
+              cntNewRiver = cntNewRiver + 1;
+            }
+            if (rsBrouter.getString("forest_class") != null) {
+              map.put("estimated_forest_class", rsBrouter.getString("forest_class"));
+              cntNewForest = cntNewForest + 1;
+            }
+
+            if (rsBrouter.getString("town_class") != null) {
+              map.put("estimated_town_class", rsBrouter.getString("town_class"));
+              cntNewTown = cntNewTown + 1;
+            }
+
+            if (rsBrouter.getString("traffic_class") != null) {
+              map.put("estimated_traffic_class", rsBrouter.getString("traffic_class"));
+              cntNewTraffic = cntNewTraffic + 1;
```

Comment:
Lots of copy-pase code. We should store source tags and BRouter tags in a map and just lookup the mapping. Same would work for those counters.